### ISSUE TITLE
`0.14` Fails to build on aarch64

### ISSUE THREAD

**Issue by fulmicoton** [nix-rust/nix#1080]
When compiling nix -0.14, I get the following error on `aarch64-linux-android`.

```
error[E0308]: mismatched types
   --> /cargo/registry/src/github.com-1ecc6299db9ec823/nix-0.14.0/src/sys/signal.rs:495:41
    |
495 |             SigHandler::SigAction(_) => (flags | SaFlags::SA_SIGINFO).bits(),
    |                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected i32, found u32
error[E0308]: mismatched types
   --> /cargo/registry/src/github.com-1ecc6299db9ec823/nix-0.14.0/src/sys/signal.rs:505:37
    |
505 |         SaFlags::from_bits_truncate(self.sigaction.sa_flags)
    |                                     ^^^^^^^^^^^^^^^^^^^^^^^ expected u32, found i32
error: aborting due to 2 previous errors
```

**Comment by asomers** [comment]
This was caused by a recent change in libc.  It's already been fixed by https://github.com/nix-rust/nix/commit/3c992d400af6162c8ac39de11d5750180b857322 and will be included in 0.14.1.  As a workaround you can either use the master branch of nix, or else limit libc to version <= 0.2.55 .

**Comment by asomers** [comment]
Should be fixed in 0.14.1, just published.
